### PR TITLE
loading external sealevel curve from private database

### PR DESCRIPTION
**Pull request by xyl96**
extract sealevel curve.txt from private database with token (using url). Only temporary solution before figuring out how to use OSF packages, because using URL is not very convenient.

### CODE REVIEW

File: examples/sealevel-curve-extraction.jl [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+#module Import_SL
+
+using Interpolations
+using DataFrames
+using CSV
+using HTTP
+#import OpenScienceFramework as OSF
+
+function sealevel(url)
+    #=osf = OSF.Client(; token = alexa)  # put your OSF token here
```
Comment (EmiliaJarochowska) [inline examples/sealevel-curve-extraction.jl:11]: This actually would work, provided the following:

1. There is a view-only link for download for the project
2. Files are in a folder within the project

For example, this would download the first file in the first folder:
```
import OpenScienceFramework as OSF

osf = OSF.Client(; token="$TOKEN")  
proj = OSF.project(osf; title="CarboKitten")
dirs = readdir(OSF.Directory, proj)

files = readdir(OSF.File, first(dirs))

url = OSF.url(first(files))
download(url)
```
But it does not recover the file name, admittedly, and doesn't solve the problem with using the token.

File: examples/cap-extSL.jl [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# ~/~ begin <<docs/src/ca-with-production.md#examples/cap-extSL.jl>>[init]
+using CarboKitten.CaProd
+using DataFrames
+using CSV
+using Interpolations
+#datasl = CSV.read("data/bs92-sealevel-curve.csv", DataFrame, header=1)
+
+DEFAULT_INPUT = CaProd.Input(
+    sea_level = t -> CSV.read("data/all-sealevel/sl.csv", DataFrame, header=1)[!,1][round(Int,t*1000)+1],
```
Comment (jhidding) [inline examples/cap-extSL.jl:9]: This amounts to a nearest neighbour interpolation. It's probably better to use a linear interpolator from the `Interpolations` package.

File: examples/cap-extSL.jl [hunk 3]
```diff
@@ -0,0 +1,26 @@
+# ~/~ begin <<docs/src/ca-with-production.md#examples/cap-extSL.jl>>[init]
+using CarboKitten.CaProd
+using DataFrames
+using CSV
+using Interpolations
+#datasl = CSV.read("data/bs92-sealevel-curve.csv", DataFrame, header=1)
```
Comment (jhidding) [inline examples/cap-extSL.jl:6]: We use version control to keep track of changes, please don't use comments to hide old code.

File: examples/sealevel-curve-extraction.jl
[the remark this answers is not in the document]
Comment (xyl96) [reply examples/sealevel-curve-extraction.jl:1]: used $name

File: examples/sealevel-curve-extraction.jl [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    url = OSF.url(file)
+    =#
+    headers = Dict("Authorization" => "Bearer $token")
+    response = HTTP.get(url,headers)# copy paste Sealelvel data from a database
+    data = split(String(response.body), "\n")
+    #data_float64 = parse.(Float64, data)
+    CSV.write("data/all-sealevel/sl.csv", DataFrame(sl=data))
+end
+
+#url = "https://raw.githubusercontent.com/xyl96/sealevel_curve/main/stoch1" # extracting from an open url
```
[the remark this answers is not in the document]
Comment (xyl96) [reply examples/sealevel-curve-extraction.jl]: ok

File: examples/sealevel-curve-extraction.jl (hunk 1 above)
Comment (xyl96) [reply examples/sealevel-curve-extraction.jl]: used datahugger and seems work now. However, one of the trials tonight could not properly extract the file, returned an empty file.

File: examples/cap-extSL.jl (hunk 3 above)
Comment (xyl96) [reply examples/cap-extSL.jl:6]: thx!

Comment (EmiliaJarochowska) [reply examples/cap-extSL.jl]: You still have comments in the current version

File: examples/plot-cap-extSL.jl [hunk 5]
```diff
@@ -1,13 +1,18 @@
...
 using CarboKitten.Visualization
 using GLMakie
+import OpenScienceFramework as OSF
+
 
 function main()
     f = Figure()
     plot_crosssection(f[1, 1], "data/ca-extSL.h5")
-    save("docs/src/fig/b13-crosssection-ext.png", f)
+    save("C:/Users/Liu00141/OneDrive - Universiteit Utrecht/Desktop/work_by_week/external plots/b13-crosssection-ext.png", f)
```
Comment (EmiliaJarochowska) [inline examples/plot-cap-extSL.jl:10]: You can't use an absolute path on your computer

File: examples/sealevel-curve-extraction.jl
Comment (EmiliaJarochowska) [inline examples/sealevel-curve-extraction.jl:1]: 1. Shall it save `data` outside of the GH repo? I.e. in `../data`? Or you add this folder to `.gitignore`.
2. Why do we need files in `.txt` and `.csv`? Can we just use one format? If using `txt` directly is a problem, I can just change the R function that produces them to save `csv`.

File: examples/cap-extSL.jl [hunk 6]
```diff
@@ -23,4 +30,5 @@ DEFAULT_INPUT = CaProd.Input(
 )
 
 CaProd.main(DEFAULT_INPUT, "data/ca-extSL.h5")
-# ~/~ end
\ No newline at end of file
+# ~/~ end
+name = "Auto000_Allo000_Stoch100V2"
```
Comment (EmiliaJarochowska) [inline examples/cap-extSL.jl:34]: why is this here?

File: examples/cap-extSL.jl [hunk 7]
```diff
@@ -3,10 +3,17 @@ using CarboKitten.CaProd
...
+    data = DataFrame(CSV.File(filepath))
+    data = hcat(collect(0:length(data.sl).-1)./1000, data.sl)
+    x = linear_interpolation(data[:,1], data[:,2])
+    return x(t)
+end
+
+#CSV.read("data/all-sealevel/$name.csv", DataFrame, header=1)[!,1][round(Int,t*1000)+1]
 DEFAULT_INPUT = CaProd.Input(
-    sea_level = t -> CSV.read("data/all-sealevel/sl.csv", DataFrame, header=1)[!,1][round(Int,t*1000)+1],
+    sea_level = t -> sealevel_curve(t,"data/all-sealevel/sl1.csv"),
```
Comment (EmiliaJarochowska) [inline examples/cap-extSL.jl:16]: It does not make much sense to download all the SL files and then run this by changing the file name by hand. I think there are several ways to change it, probably something to discuss with @jhidding

**Review by EmiliaJarochowska — CHANGES_REQUESTED** [review]
1. Please resolve the comments which have been fixed
2. Remove old unused code, don't leave it in comments, you can always come back to a previous version since we have version control. You might want to split your commits in such a way that it is possible to do it atomically?
3. Never leave an absolute path to files in your local file system
4. Watch out for Entangled tags, if you're not sure what I mean, please have a look at Entangled webpage, it has a nice intro! or ask @jhidding :-)

File: examples/cap-extSL.jl (hunk 7 above)
Comment (xyl96) [reply examples/cap-extSL.jl]: now it can run all the files with their names on it

File: examples/cap-extSL.jl (hunk 6 above)
Comment (xyl96) [reply examples/cap-extSL.jl]: ok

File: examples/sealevel-curve-extraction.jl
Comment (xyl96) [reply examples/sealevel-curve-extraction.jl:1]: the original datafile is in txt so I first import txt and then translate into csv in julia. It's easier to use csv files actually.

File: examples/plot-cap-extSL.jl (hunk 5 above)
Comment (xyl96) [reply examples/plot-cap-extSL.jl:10]: my bad here
